File: leet/0061-rotate-list/rotate-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func rotateRight(head *ListNode, k int) *ListNode {
    if head == nil || head.Next == nil { return head}

    fast := head
	slow := head
	
	for ;k > 0; k-- {
		fast = fast.Next
		if fast == nil {
			fast = head
		}
	}
	
	if fast == head {return head}
	
	for fast.Next != nil {
		fast = fast.Next
		slow = slow.Next
	}
	
	newHead := slow.Next
	slow.Next = nil
	fast.Next = head
	
	return newHead
}

File: leet/0061-rotate-list/rotate-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
var rotateRight = function(head, k) {
    if (!head || !head.next) return head
    let list = []
    
    while(head){
        list.push(head)
        head = head.next
    }
    
    let mov = k % list.length
    if(mov == 0) return list[0] 
    
    list[list.length - mov - 1].next = null
    list[list.length - 1].next = list[0]
    return list[list.length - mov]
};
